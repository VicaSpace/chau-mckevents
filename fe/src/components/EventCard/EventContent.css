
.event-date {
    width: 154.72px;
    height: 57.12px;
    margin-top: 11.53px;
    margin-left: 30.36px;
}

.event-date-month {
    font-family: 'Inter';
    font-style: normal;
    font-weight: 600;
    font-size: 32px;
    line-height: 39px;

    color: #263238;
    text-align: center;
}

.event-year {
    text-align: center;

    font-family: 'Inter';
    font-style: normal;
    font-weight: 600;
    font-size: 15px;
    line-height: 18px;
}

.event-time {
    width: 301px;
    height: 14px;
    left: 254px;
    margin-top: 0px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 14px;
    /* identical to box height */

    color: #000000;
    position: relative;
}

.event-location {
    width: 336px;
    height: 42px;
    left: 255px;
    margin-top: 5px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 14px;

    color: #000000;
    position: relative;
}

.event-content {
    display: flex;
    flex-direction: row;

    position: absolute;
}