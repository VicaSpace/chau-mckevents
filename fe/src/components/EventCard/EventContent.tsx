import moment from 'moment';
import React from 'react';
import './EventContent.css';

const EventContent: React.FC<{
  date: string;
  time: string;
  location: string;
  timeSuggestions: any[];
  status: string;
}> = ({ date, time, location, timeSuggestions, status }) => {
  const renderTime = () => {
    if (status === 'CONFIRMED') {
      return <>Time: {moment(time).format('HH:MM').toString()}</>;
    } else if (status === 'NOT_CONFIRMED') {
      return (
        <>
          Time suggestions:{' '}
          {timeSuggestions
            .map((suggestion) =>
              moment(suggestion.time).format('HH:MM').toString()
            )
            .join(' / ')}
        </>
      );
    }
  };

  return (
    <div className="card-content" data-testid="test-card-content">
      <div className="event-date">
        <div className="event-date-month">
          {moment(date).format('MMMM Do').toString()}
        </div>
        <div className="event-year">
          {moment(date).format('YYYY').toString()}
        </div>
      </div>
      <div className="event-time" data-testid="test-event-time">
        {renderTime()}
      </div>
      <div className="event-location" data-testid="test-event-location">
        Location: {location}
      </div>
    </div>
  );
};

export default EventContent;
