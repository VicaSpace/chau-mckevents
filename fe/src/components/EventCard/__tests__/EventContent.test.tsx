import {render, screen} from '@testing-library/react'
import React from 'react'
import EventContent from '../EventContent';

const dummyProps = {
    date: '2020-02-02 00:00:00',
    time: '2020-02-02 08:00:00',
    location: 'test',
    timeSuggestions: [
        {
            id: 1,
            time: '2020-02-02 08:00:00',
        },
        {
            id: 1,
            time: '2020-02-02 09:00:00',
        },
    ]
}

describe('Test: Event Card component screenshot', () => {
    it('should render the Event Card correctly', () => {
      const { container } = render(
          <EventContent {...dummyProps} status='NOT_CONFIRMED' />
      );
      expect(container).toMatchSnapshot();
    });
  });

describe('Test: Event Card props NOT_CONFIRMED', () => {
    it('should render location node', () => {
        render(<EventContent {...dummyProps} status='NOT_CONFIRMED' />);
        const eventLocation = screen.getByTestId('test-event-location');
        expect(eventLocation.classList).toContain('event-location');
        expect(eventLocation.textContent).toContain(dummyProps.location);
    });

    it('should render time node', () => {
        render(<EventContent {...dummyProps} status='NOT_CONFIRMED'/>);
        const eventLocation = screen.getByTestId('test-event-time');
        expect(eventLocation.classList).toContain('event-time');
        expect(eventLocation.textContent).toContain('Time suggestions');
        // expect(eventLocation.textContent).toContain('08:00 / 09:00');
    });
});


describe('Test: Event Card props CONFIRMED', () => {
    it('should render location node', () => {
        render(<EventContent {...dummyProps} status='CONFIRMED' />);
        const eventLocation = screen.getByTestId('test-event-location');
        expect(eventLocation.classList).toContain('event-location');
        expect(eventLocation.textContent).toContain(dummyProps.location);
    });

    it('should render time node', () => {
        render(<EventContent {...dummyProps} status='CONFIRMED'/>);
        const eventLocation = screen.getByTestId('test-event-time');
        expect(eventLocation.classList).toContain('event-time');
        expect(eventLocation.textContent).toContain('Time:');
        // expect(eventLocation.textContent).toContain('08:00 / 09:00');
    });
});

