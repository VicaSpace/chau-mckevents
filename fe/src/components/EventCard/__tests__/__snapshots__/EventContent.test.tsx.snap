// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Event Card component screenshot should render the Event Card correctly 1`] = `
<div>
  <div
    class="card-content"
    data-testid="test-card-content"
  >
    <div
      class="event-date"
    >
      <div
        class="event-date-month"
      >
        February 2nd
      </div>
      <div
        class="event-year"
      >
        2020
      </div>
    </div>
    <div
      class="event-time"
      data-testid="test-event-time"
    >
      Time suggestions:
       
      08:02 / 09:02
    </div>
    <div
      class="event-location"
      data-testid="test-event-location"
    >
      Location: 
      test
    </div>
  </div>
</div>
`;
